src/org/opensolaris/opengrok/util/OptionParser.java

```diff
@@ -0,0 +1,701 @@
+                    
+                    if (!converters.containsKey(opt.valueType)) {
+                        throw new ParseException(
+                            "No conversion handler for data type " + opt.valueType, ii);
+                    }
+                    
+                    try {
+                        DataParser data = converters.get(opt.valueType);
+                        value = data.converter.apply(opt.value);
+                        
+                    } catch (Exception e) {
+                        throw new ParseException(
+                           "Failed to parse ("+opt.value+") as value of " + opt.names, ii);
+                    }
+                }
+                
+                if (opt.action != null) {
+                    opt.action.accept(value); // 'do' assigned action
+                }
+                optind = ++ii;
+            }
+        }
+        
+        // Prepare to gather any remaining arguments
+        // to send back to calling program.
+        
+        String[] remainingArgs = null;
+        
+        if (optind == -1) {
+            remainingArgs = args;
+        } else if (optind < args.length) {
+            remainingArgs = Arrays.copyOfRange(args, optind, args.length);
+        }
+        
+        return remainingArgs;
+    }
+    
+    private String program() {
+        // Locate main program name from stack trace.
+        StackTraceElement[] stack = Thread.currentThread ().getStackTrace ();
+        StackTraceElement main = stack[stack.length - 1];
+        String[] name = main.getClassName().split("\\.");
+        return name[name.length-1];
+    }
+    
+    private String getPrologue() {
+        // Assign default prologue statement when none given.
+        if (prologue == null) {
+            prologue = "Usage: " + program() + " [options]";
+        }
+        
+        return prologue;
+    }
+    
+    /**
+     * Place text in option summary
+     * @param text to be inserted into option summary.
+     * 
+     * Example usage:
+     *  OptionParser opts = OptionParser.Do( parser -&gt; {
+     *   
+     *    parser.prologue = String.format("Usage: %s [options] bubba smith", program);
+     *    parser.separator("");
+     * 
+     *    parser.on("-y value", "--why me", "This is a description").Do( v -&gt; {
+     *        System.out.println("got " + v);
+     *    });
+     *       
+     *    parser.separator("  ----------------------------------------------");
+     *    parser.separator("  Common Options:");
+     *    ...
+     * 
+     *    parser.separator("  ----------------------------------------------");
+     *    parser.epilogue = "  That's all Folks!";
+     */
+    public void separator(String text) {
+        usageSummary.add(text);
+    }
+    
+    /**
+     * Obtain option summary
+     * @return option summary
+     */
+    public String getUsage() {
+        
+        StringWriter wrt = new StringWriter();
+        String indent = "  ";
```

Comment:
would be nice if the indent was configurable

Reply:
Would another method getUsage(indent) satisfy this itch?

Reply:
I think so.